src/test/java/gift/ProductControllerTest.java

```diff
@@ -0,0 +1,114 @@
+package gift;
+
+import gift.dto.Product;
+import gift.service.ProductService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+import static org.hamcrest.Matchers.*;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@Transactional
+public class ProductControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;  //Spring MockMvc 프레임워크를 사용하여 HTTP 요청 및 응답 테스트
+
+    @Autowired
+    private ProductService productService;
+
+    private Product sampleProduct;
+
+
+    @BeforeEach
+    void setUp() {
+        sampleProduct = new Product();
+        sampleProduct.setName("아이스 카페 아메리카노 T");
+        sampleProduct.setPrice(4500L);
+        sampleProduct.setImageUrl("https://st.kakaocdn.net/product/gift/product/20231010111814_9a667f9eccc943648797925498bdd8a3.jpg");
+        productService.addProduct(sampleProduct);
+    }
```

Comment:
```suggestion
    sampleProduct = new Product(null, "아이스 카페 아메리카노 T", 4500L,
                "https://st.kakaocdn.net/product/gift/product/20231010111814_9a667f9eccc943648797925498bdd8a3.jpg");
        productService.addProduct(sampleProduct);
```

이런 방식으로 사용할 수도 있을 것 같은데요.!
굳이 setter를 사용한 이유가 있을까요?

+ beforeEach는 테스트 메서드의 수만큼 해당 작업을 반복하게 되는데요, sampleProduct를 매번 새로 만들 필요는 없을 것 같아요!

Reply:
setter를 평소에 자주 써서 편한대로 작성하게 된 것 같습니다 😅

피드백 주신 내용처럼 코드 수정했습니다!